ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/142__distributed_workload_components.robot

```diff
@@ -0,0 +1,114 @@
+*** Settings ***
+Documentation       Test Cases to verify DSC Distributed Workloads Components
+
+Library             Collections
+Resource            ../../../../Resources/OCP.resource
+Resource            ../../../../Resources/ODS.robot
+Suite Setup         Suite Setup
+Suite Teardown      Suite Teardown
+
+
+*** Variables ***
+${OPERATOR_NS}                  ${OPERATOR_NAMESPACE}
+${APPLICATIONS_NS}              ${APPLICATIONS_NAMESPACE}
+${DSC_NAME}                     default-dsc
+${KUEUE_LABEL_SELECTOR}         app.kubernetes.io/name=kueue
+${KUEUE_DEPLOYMENT_NAME}        kueue-controller-manager
+${CODEFLARE_LABEL_SELECTOR}     app.kubernetes.io/name=codeflare-operator
+${CODEFLARE_DEPLOYMENT_NAME}    codeflare-operator-manager
+${RAY_LABEL_SELECTOR}           app.kubernetes.io/name=kuberay
+${RAY_DEPLOYMENT_NAME}          kuberay-operator
+${IS_PRESENT}    0
+${IS_NOT_PRESENT}    1
+
+
+*** Test Cases ***
+Validate Kueue Managed State
+    [Documentation]    Validate that the DSC by default sets component 'kueue' to sate Managed,
+    ...    check that kueue deployment and pod are created
+    [Tags]    Operator    Tier1    RHOAIENG-5435    kueue-managed
+
+    Check Component Resources In Managed State   kueue    ${KUEUE_DEPLOYMENT_NAME}    ${KUEUE_LABEL_SELECTOR}
+
+Validate Kueue Removed State
+    [Documentation]    Validate that Kueue management state Removed does remove relevant resources.
+    [Tags]    Operator    Tier1    RHOAIENG-5435    kueue-removed
+
+    Set DSC Component Removed State And Wait For Completion   kueue    ${KUEUE_DEPLOYMENT_NAME}    ${KUEUE_LABEL_SELECTOR}
+
+    [Teardown]     Restore DSC Component Managed State And Wait For Completion    kueue    ${KUEUE_DEPLOYMENT_NAME}    ${KUEUE_LABEL_SELECTOR}
```

Comment:
in order to avoid mutating the already present cluster configuration, we should restore to the original state. So, the flow being:

1. fetch component status
2. set the desired component status
3. restore the original component status (step 1)

Reply:
Resolved